Modules/_sre/sre_lib.h

```diff
@@ -560,13 +560,28 @@ typedef struct {
     Py_ssize_t last_ctx_pos;
 } SRE(match_context);
 
-#define MAYBE_CHECK_SIGNALS                                        \
+#define _MAYBE_CHECK_SIGNALS                                       \
     do {                                                           \
         if ((0 == (++sigcount & 0xfff)) && PyErr_CheckSignals()) { \
             RETURN_ERROR(SRE_ERROR_INTERRUPTED);                   \
         }                                                          \
     } while (0)
 
+#ifdef Py_DEBUG
+# define MAYBE_CHECK_SIGNALS                                       \
+    do {                                                           \
+        _MAYBE_CHECK_SIGNALS;                                      \
+        if (state->fail_after_count >= 0) {                        \
+            if (state->fail_after_count-- == 0) {                  \
```

Comment:
Setting `state->fail_after_count` does not affect subsequent calls of the `Pattern` methods, but it still affects the `finditer()` iterator which reuses state between iterations.

Reply:
I deleted my previous reply. 
Maybe setting `state->fail_after_count = -1` doesn't make sense, it's better to make the `finditer()` iterator stop subsequent matchings.